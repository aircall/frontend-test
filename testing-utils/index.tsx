import React from "react";
import "jest-styled-components";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { MemoryRouter } from "react-router-dom";

import { theme } from "../src/common/theme";

type Options = RenderOptions & {
  pathName?: string;
};

const customRender = (
  ui: React.ReactElement,
  { pathName, ...options }: Options = {}
) => {
  const AllTheProviders = ({ children }: { children?: React.ReactNode }) => {
    return (
      <ThemeProvider theme={theme}>
        <MemoryRouter
          initialEntries={[{ pathname: pathName || "/", key: "testKey" }]}
          initialIndex={0}
        >
          {children}
        </MemoryRouter>
      </ThemeProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
