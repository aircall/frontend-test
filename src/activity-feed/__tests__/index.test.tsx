import React from "react";

import { render } from "testing-utils";

import { ActivityFeed } from "..";

describe("activity-feed", () => {
  describe("ActivityFeed", () => {
    it("should render correctly", () => {
      const { container } = render(<ActivityFeed />);

      expect(container).toMatchSnapshot();
    });
  });
});
