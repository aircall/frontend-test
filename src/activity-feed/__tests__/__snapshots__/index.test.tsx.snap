// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity-feed ActivityFeed should render correctly 1`] = `
<div>
  Some activities should be here
</div>
`;
